src/NumericalAlgorithms/DiscontinuousGalerkin/Actions/InterfaceActionHelpers.hpp

```diff
@@ -10,89 +10,80 @@
 #include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
 #include "Domain/Direction.hpp"
 #include "Domain/FaceNormal.hpp"
+#include "Domain/InterfaceHelpers.hpp"
 #include "Domain/Mesh.hpp"
 #include "Domain/Tags.hpp"
 #include "NumericalAlgorithms/DiscontinuousGalerkin/FluxCommunicationTypes.hpp"
 #include "NumericalAlgorithms/DiscontinuousGalerkin/MortarHelpers.hpp"
 #include "NumericalAlgorithms/DiscontinuousGalerkin/Tags.hpp"
 
 namespace DgActions_detail {
+
 template <typename Metavariables, typename DataBoxType, typename DirectionsTag,
-          typename NumericalFlux>
-auto compute_packaged_data(
+          typename NumericalFlux, size_t VolumeDim = DirectionsTag::volume_dim,
+          typename FluxCommTypes = dg::FluxCommunicationTypes<Metavariables>,
+          typename PackagedData = typename FluxCommTypes::PackagedData>
+DirectionMap<VolumeDim, PackagedData> compute_packaged_data(
     const DataBoxType& box,
-    const Direction<Metavariables::system::volume_dim>& direction,
     const NumericalFlux& normal_dot_numerical_flux_computer,
-    const DirectionsTag /*meta*/, const Metavariables /*meta*/) noexcept ->
-    typename dg::FluxCommunicationTypes<Metavariables>::PackagedData {
-  constexpr size_t volume_dim = Metavariables::system::volume_dim;
-
-  using flux_comm_types = dg::FluxCommunicationTypes<Metavariables>;
-
-  using package_arguments =
-  typename Metavariables::normal_dot_numerical_flux::type::argument_tags;
-
-  const auto &face_mesh =
-      db::get<Tags::Interface<DirectionsTag, Tags::Mesh<volume_dim - 1>>>(box)
-          .at(direction);
-
-  return db::apply<tmpl::transform<
-      package_arguments, tmpl::bind<Tags::Interface, DirectionsTag, tmpl::_1>>>(
-      [&face_mesh, &direction,
-       &normal_dot_numerical_flux_computer ](const auto&... args) noexcept {
-        typename flux_comm_types::PackagedData ret(
-            face_mesh.number_of_grid_points(), 0.0);
-        normal_dot_numerical_flux_computer.package_data(make_not_null(&ret),
-                                                        args.at(direction)...);
-        return ret;
+    const DirectionsTag /*meta*/, const Metavariables /*meta*/) noexcept {
+  return interface_apply<
+      DirectionsTag,
+      tmpl::flatten<tmpl::list<::Tags::Mesh<VolumeDim - 1>,
+                               typename NumericalFlux::argument_tags>>,
+      InterfaceHelpers_detail::get_volume_tags<NumericalFlux>>(
```

Comment:
Move `get_volume_tags` out of the detail namespace so you're not accessing private stuff here.  I think that function is clear/useful enough for general use.